### PR TITLE
Fix/8702 hash passwords

### LINKED ISSUE #8702: Unable to sign in  or restore account if password cointains ' symbol (quotemark)
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug
*Summary*: using password with quotemark `'` leads to error:
```
08-05 06:24:58.791  4332  4732 E GoLog   : failed to set key pragma
08-05 06:24:58.796  4332  4391 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :multiaccounts.login.callback/login-success
08-05 06:24:58.810  4332  4391 E ReactNativeJS: ERROR [status-im.multiaccounts.login.core:125] - failed to login failed to set key pragma
08-05 06:24:58.820  4332  4392 E unknown:ReactNative: console.error: "ERROR [status-im.multiaccounts.login.core:125] - failed to login failed to set key pragma", stack:
08-05 06:24:58.820  4332  4392 E unknown:ReactNative: error@345.js:96:29
08-05 06:24:58.820  4332  4392 E unknown:ReactNative: hca@7201:45
08-05 06:24:58.820  4332  4392 E unknown:ReactNative: a@7303:53
08-05 06:24:58.820  4332  4392 E unknown:ReactNative: c@91168:136
08-05 06:24:58.820  4332  4392 E unknown:ReactNative: b@91163:24
08-05 06:24:58.820  4332  4392 E unknown:ReactNative: <unknown>@33320:26
08-05 06:24:58.820  4332  4392 E unknown:ReactNative: <unknown>@33605:52
08-05 06:24:58.820  4332  4392 E unknown:ReactNative: Jb@11905:35
08-05 06:24:58.820  4332  4392 E unknown:ReactNative: gc@3075:47
```
 and user automatically logged out

#### Expected behavior
can use password with  ' symbol

#### Actual behavior
error

### Reproduction

- Open Status
- Try to create account with password that contains `'`

### Additional Information

[comment]: # (Please do your best to fill this out.)
- Status version: nigthly 04/08/2019. 
- Operating System: Android, iOS

#### Logs

Full test and logcat can be found [here](https://app.saucelabs.com/tests/b6eec76897c7489a871cef95ca8b53cf#30)

### PR DESCRIPTION
**Pull request by yenda**
### Summary

fix #8702

clean up native-module

- remove impl/modules.cljs namespace because it was needlessly redundant
- remove `create-account` unused call
- add documentation for some calls
- only keep `(status)` check for relevant calls that are failing tests
because they are evaluated

- use ethereum.utils sha3 function to hash passwords before sending them
to status-go
- some native calls take password as one of their params: they now take
hashed-password (still a string, only relevant for status-react caller)
- some native calls take password within a map of rpc params: it now
needs to be hashed with `ethereum.core/sha3`

#### Areas that maybe impacted
Any area that takes a password as input

### Steps to test

Test 
- login
- recovery
- adding account
- signing message
- signing transaction

changes are not platform dependent so it is not a big deal if iOS doesn't build
(I tested all of these on Android emulator on Ropsten with simpledapp.eth)

No password should be logged

status: ready

### CODE REVIEW

**Review by vitvly — APPROVED** [review]

File: src/status_im/multiaccounts/create/core.cljs [hunk 1]
```diff
@@ -50,8 +44,9 @@
   [{:keys [db] :as cofx}]
   (let [{:keys [selected-id address key-code]} (:intro-wizard db)
         {:keys [address]} (get-selected-multiaccount cofx)
+        hashed-password (ethereum/sha3 (security/safe-unmask-data key-code))
         callback #(re-frame/dispatch [::store-multiaccount-success key-code])]
-    {::store-multiaccount [selected-id address key-code callback]}))
+    {::store-multiaccount [selected-id address hashed-password callback]}))
```
Comment (flexsurfer) [inline src/status_im/multiaccounts/create/core.cljs:49]: maybe better to unmask it in fx ?

File: src/status_im/native_module/core.cljs [hunk 2]
```diff
@@ -1,130 +1,262 @@
 (ns status-im.native-module.core
-  (:require [status-im.native-module.impl.module :as native-module]))
+  (:require [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [status-im.react-native.js-dependencies :as rn-dependencies]
+            [status-im.ui.components.react :as react]
+            [status-im.utils.platform :as platform]
+            [status-im.utils.types :as types]))
+
+(defn status []
```
Comment (flexsurfer) [inline src/status_im/native_module/core.cljs:9]: why is it a function?

File: src/status_im/native_module/impl/module.cljs [hunk 3]
```diff
@@ -1,236 +0,0 @@
-(ns status-im.native-module.impl.module
```
Comment (flexsurfer) [inline src/status_im/native_module/impl/module.cljs:1]: we needed this for tests right? how do tests work now?

Comment (yenda) [reply src/status_im/native_module/impl/module.cljs:1]: it was never used for tests, and there is only 2 functions that are not in an effect and hence called during tests, it is those whis (when (status)) guard

File: src/status_im/native_module/core.cljs (hunk 2 above)
Comment (yenda) [reply src/status_im/native_module/core.cljs:9]: I guess it is for tests

File: src/status_im/multiaccounts/create/core.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/multiaccounts/create/core.cljs:49]: fxs are suppose to do as little as possible besides the side effect itself, what would be the advantage?

File: src/status_im/native_module/core.cljs (hunk 2 above)
Comment (yenda) [reply src/status_im/native_module/core.cljs:9]: might be worth moving it to status-im.react-native.js-dependencies at some point I'll add a comment

**Comment by churik** [comment]
@yenda 
I've almost finished and conflicts appeared. 
Can you please resolve them and elaborate on what areas should be retested after resolving?

**Comment by yenda** [comment]
@churik it was just the parameters of connection-change, if you want to test it just switch from wifi to mobile network. It was a pretty simple merge conflict to solve though

**Comment by churik** [comment]
@yenda 
App with last commit https://github.com/status-im/status-react/commit/e18cb988665ebde55421d7179aa2eb7cef33d4b8 crashes on start (Android 8):
photo_2019-09-05 16 15 21


Tested on Android 8 on build https://github.com/status-im/status-react/commit/196f88285748e7c23ba668b14c5f57420519170c:
- passwords with `-/:;()$&@“.,?!‘[]{}#%^*+=_\|~<>€£¥•.,?!’!@#$%^&*()[{:"|\<>?,./~`±§}]`, some arabic and Spanish symbols on login, recovery, adding account, signing message, signing transaction in wallet and dapp
- password validation (short password, wrong passwrd)
- password is not logged in "Status.log", "geth.logv"

**Comment by yenda** [comment]
@churik the `expensives?` param in connection-change had to be casted to a boolean, it was done in the  deleted module namespace so I missed it when merging. Should be good now

**Pull request merged**